core-bundle/src/Resources/public/core.js

```diff
@@ -992,10 +994,53 @@ var Backend =
 	},
 
 	/**
-	 * Get the current scroll offset and store it in a cookie
+	 * Store the current scroll offset in sessionStorage
 	 */
 	getScrollOffset: function() {
-		document.cookie = "BE_PAGE_OFFSET=" + window.getScroll().y + "; path=" + (Contao.path || '/');
+		window.sessionStorage.setItem('contao_backend_offset', parseInt(window.getScroll().y, 10));
+	},
+
+	/**
+	 * Remove the legacy BE_PAGE_OFFSET cookie, scroll to the current offset if
+	 * it was defined and add the "down" CSS class to the header.
+	 */
+	initScrollOffset: function() {
+		// Kill the legacy cookie here; this way it can be sent by the server
+		// but it wont be resent by the client in the next request
+		Cookie.dispose('BE_PAGE_OFFSET');
+
+		// Add events to the submit buttons so they can reset the offset
+		// (except for "save", which always stays on the same page)
+		$$('.tl_submit_container button[name][name!="save"]').each(function(button) {
+			var form = button.getParent('form');
+			if (!form) return;
+
+			button.addEvent('click', function(e) {
```

Comment:
That looks reeeally hacky. If I understand correctly, we need to add the hidden element to keep track of which button was clicked? And simply handling the click event would submit "too fast" for the JS to apply?
Maybe we could use "mouseup" or similar instead? @ausi

Reply:
Wouldn’t the following code achieve the same?

```js
button.addEvent('click', function(e) {
	sessionStorage.removeItem('contao_backend_offset');
}
```

If you don’t call `preventDefault()` you also don’t habe to manually call `form.submit()` and the clicked button should be handled as usual?

Reply:
I've tested if you don't call `preventDefault()` it just does what the default is: submit the form. It doesn't do anything else.

Reply:
I was not able to reproduce that, the value was successfully removed from the sessionStorage in my test.

Reply:
I'll double check. It was the reason why I built this complicated solution in the first place.

Reply:
Strange, it works for me too now. Removed complexity in 3535902d9f4aadea6832072e2ae6c8c49e752f0e